sqlobject/src/main/java/org/jdbi/v3/sqlobject/mixins/SqlObjectMixin.java

```diff
@@ -16,20 +16,34 @@
 import org.jdbi.v3.core.Handle;
 import org.jdbi.v3.core.HandleCallback;
 import org.jdbi.v3.core.HandleConsumer;
+import org.jdbi.v3.core.HandleSupplier;
 
 /**
- * A mixin interface to define a method for obtaining the Handle attached to a sql object
+ * SqlObject base mixin interface.  Custom mixins may be written as subinterfaces
+ * of this class with default methods.
  */
-public interface GetHandle
+public interface SqlObjectMixin<This extends SqlObjectMixin<This>>
 {
     /**
-     * Obtain the handle associated to this sql object.
-     *
-     * @return the handle
+     * @return the {@link HandleSupplier} bound to this sql object.
+     */
+    HandleSupplier getHandleSupplier();
+
+    /**
+     * @return the handle open in the current sql object context.
      */
     Handle getHandle();
```

Comment:
The more I think about it, the more I question the value of mixins.  The only one we have in core can be rewritten as default methods trivially, and I'm having trouble thinking of a use case.  If you want to implement arbitrary methods you have `@SqlMethodHannotation`.  If you want to implement a single method, you can just make it default.